test/SVMSpokePoolClient.fills.ts

```diff
@@ -221,48 +225,44 @@ describe("SVMSpokePoolClient: Fills", function () {
   });
 
   it("Closes the fill pda after the fill deadline has passed", async () => {
-    const currentSlot = await solanaClient.rpc.getSlot({ commitment: "confirmed" }).send();
-    const currentSlotTimestamp = await solanaClient.rpc.getBlockTime(currentSlot).send();
-    const fillDeadline = Number(currentSlotTimestamp) + 1;
-    await setCurrentTime(signer, solanaClient, Number(currentSlotTimestamp));
-    const newRelayData = { ...relayData, depositId: new Uint8Array(intToU8Array32(getRandomInt())), fillDeadline };
+    const provider = solanaClient.rpc;
+    let { timestamp } = await findNearestTime(provider);
+
+    await setCurrentTime(signer, solanaClient, timestamp);
+    const newRelayData = {
+      ...relayData,
+      depositId: new Uint8Array(intToU8Array32(getRandomInt())),
+      fillDeadline: timestamp + 1,
+    };
     const formattedRelayData = formatRelayData(newRelayData);
     await mintTokens(signer, solanaClient, mint.address, BigInt(relayData.outputAmount));
-    const { fillInput, relayData: fillRelayData } = await sendCreateFill(
-      solanaClient,
-      signer,
-      mint,
-      decimals,
-      newRelayData
-    );
+    const {
+      fillInput,
+      relayData: { fillDeadline },
+    } = await sendCreateFill(solanaClient, signer, mint, decimals, newRelayData);
+    expect(fillDeadline >= timestamp + 1).to.be.true;
 
     const fillStatusAfterFill = await spokePoolClient.relayFillStatus(formattedRelayData);
     expect(fillStatusAfterFill).to.equal(FillStatus.Filled);
 
-    try {
-      const closePdaInstruction = await createCloseFillPdaInstruction(signer, solanaClient.rpc, fillInput.fillStatus);
-      await signAndSendTransaction(solanaClient, closePdaInstruction);
-    } catch (error) {
-      expect(error.context.logs.some((log) => log.includes("The fill deadline has not passed!"))).to.be.true;
-    }
+    // Verify that it's not possible to close the PDA at present.
+    let closePdaInstruction = await createCloseFillPdaInstruction(signer, solanaClient.rpc, fillInput.fillStatus);
+    await assertPromiseError(signAndSendTransaction(solanaClient, closePdaInstruction));
```

Comment:
Refactored this because it's not guaranteed that `signAndSendTransaction()` actually throws, in which case the subsequent `expect()` statement won't actually fire.